src/include/processor/operator/gds_call_shared_state.h

```diff
@@ -62,9 +58,18 @@ struct GDSCallSharedState {
     std::vector<common::NodeSemiMask*> getPathNodeMasks() const { return pathNodeMask->getMasks(); }
     NodeOffsetMaskMap* getPathNodeMaskMap() const { return pathNodeMask.get(); }
 
+    FactorizedTable* claimLocalTable(storage::MemoryManager* mm);
+    void reclaimLocalTable(FactorizedTable* table);
+    void mergeLocalTables();
+
 private:
     std::unique_ptr<NodeOffsetMaskMap> inputNodeMask = nullptr;
     std::unique_ptr<NodeOffsetMaskMap> pathNodeMask = nullptr;
+    // We implement a local ftable pool to avoid generate many small ftables when running GDS.
```

Comment:
I would remove these comments. Instead I would open an issue that explains the alternative (and I think better) solution and assign it to Guodong and Ben.

Reply:
Done